modules/process/manager.go

```diff
@@ -86,6 +86,10 @@ func (pm *Manager) AddContext(parent context.Context, description string) (ctx c
 // Most processes will not need to use the cancel function but there will be cases whereby you want to cancel the process but not immediately remove it from the
 // process table.
 func (pm *Manager) AddContextTimeout(parent context.Context, timeout time.Duration, description string) (ctx context.Context, cancel context.CancelFunc, finshed FinishedFunc) {
+	if timeout <= 0 {
```

Comment:
Seems like there are some user-defined timeouts being passed to this function, I'm not sure that in the middle of normal operations a panic should occur from a process...  

Reference: https://github.com/go-gitea/gitea/blob/a82fd98d5368a75cbcf6b74c12f58f3f81e66662/services/mailer/mailer.go#L276

Reply:
hm that would mean we either should fail on gitea init or put out a warn and ignore this setting

Reply:
@Gusted 

When passing timeout<=0:

* Before:  then the context is cancelled immediately (a non-obvious bug, which may lead to more strange problems)
* Now: a panic is reported as early as possible, when we catch the bug and can fix it.

From my experience, in a big and complex system, everything should be defined as a clear behavior, if there is something unexpected happening, the error should be reported in the first time. We should always expose mistakes, instead of hiding mistakes.

So I think this new mechanism is better and more stable, this refactoring is only done in 1.17 and we have enough time to catch more bugs in future.

And fortunately, the `SendmailTimeout` has a default value: `SendmailTimeout:     sec.Key("SENDMAIL_TIMEOUT").MustDuration(5 * time.Minute),`, so there will be no problem now.

Reply:
Or we could ignore the timeout context if timeout <= 0 but not panic

Reply:
No, ignoring non-sense values in low-level frameworks is wrong. Many Go functions also panic if there is an invalid input.

Reply:
> And when he reports the bug to gitea issues, the panic message is also very clear for maintainers.

I think we should prevent such issues from being reported. I'm not sure at which level the context cancelled are being reported to the admin, but otherwise it might be their "first time" that they observe that their configuration is incorrect. I think we should have a little note in the 1.17 announcement that a negative timeout can lead to panics and errors and that `0` should be used instead.

Reply:
> Many Go functions also panic if there is an invalid input.

Many were written without being able to return errors, and given they don't want to break backwards compatibility, they instead panic.

Reply:
> > And when he reports the bug to gitea issues, the panic message is also very clear for maintainers.
> 
> I think we should prevent such issues from being reported. I'm not sure at which level the context cancelled are being reported to the admin, but otherwise it might be their "first time" that they observe that their configuration is incorrect. I think we should have a little note in the 1.17 announcement that a negative timeout can lead to panics and errors and that `0` should be used instead.

Why do we need to mention that? If users were using `-1`, how can their system work correctly?

Reply:
> Why do we need to mention that? If users were using `-1` or `0`, how can their system work correctly?

I have no clue, but from what I've learned is that people can somehow survive with incorrect configurations without noticing.  Anyhow, approving.

Reply:
To be clear:

If a user were using `-1` or `0`, and their system works correctly, after this PR, nothing changes. Because these values are processed to defaults correctly by Gitea.

If a user were using `-1` or `0` and their system doesn't work correctly, before they know nothing, after this PR they knows. That's a improvement.